### PR TITLE
Add support for starting Notary and Vault containers on IBM LinuxONE …

### PR DESCRIPTION
**Pull request by vishwabmc**
…on premise

### CODE REVIEW

File: deployments/ibm/LinuxONE/vault/contract_generator/compose.grep11/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+services:
+  vault-releases:
+    image: de.icr.io/metaco/vault-releases@sha256:${image_sha256}
```
Comment (angelnu) [inline deployments/ibm/LinuxONE/vault/contract_generator/compose.grep11/docker-compose.yml:3]: Let us make the container repo a variable and use Azure as the default (so it is the same for on-prem and cloud)

**Review by angelnu — CHANGES_REQUESTED** [review]

**Comment by angelnu** [comment]
Thanks for the contribution @vishwabmc !

Could you please add Readme to each module to explain what it does?

We also need a Readme for how to combine the modules to deploy. Example:

1.  deploy non-secure components
2. Deploy Notary
    1. create encrypted contract (and point that the key parts must be kept safe!) - explain when to pick grep11 and when hpcs
    2. create notary with contract
3. Deploy Vault

File: deployments/ibm/LinuxONE/vault/contract_generator/compose.grep11/docker-compose.yml (hunk 1 above)
Comment (vishwabmc) [reply deployments/ibm/LinuxONE/vault/contract_generator/compose.grep11/docker-compose.yml:3]: Sure.. I will add it.

**Comment by vishwabmc** [comment]
> Thanks for the contribution @vishwabmc !
> 
> Could you please add Readme to each module to explain what it does?
> 
> We also need a Readme for how to combine the modules to deploy. Example:
> 
> 1. deploy non-secure components
> 2. Deploy Notary
>    
>    1. create encrypted contract (and point that the key parts must be kept safe!) - explain when to pick grep11 and when hpcs
>    2. create notary with contract
> 3. Deploy Vault

@angelnu Sure.. I was thinking on sharing these complete documents with the clients since I was not sure if that is okay to describe all that here. If you give a go ahead to do that here, I will add.

File: deployments/ibm/LinuxONE/vault/contract_generator/compose.grep11/docker-compose.yml (hunk 1 above)
Comment (vishwabmc) [reply deployments/ibm/LinuxONE/vault/contract_generator/compose.grep11/docker-compose.yml:3]: taking a step back: This is the auth section if we used IBM.

```
        "de.icr.io": {
          "username": "iamapikey",
          "password": var.ibm_icr_api_key
        }
```

If we make it something else, does that work ?. Also, in that case,  we need to get the confirmation from IBM to see if we can point this to Azure for example, and what are the things that are needed to reach Azure etc.

Comment (vishwabmc) [reply deployments/ibm/LinuxONE/vault/contract_generator/compose.grep11/docker-compose.yml:3]: Nevermind, I checked and the username and password still works with metaco.azurecr.io. I will just restructure the variables a bit.

**Comment by vishwabmc** [comment]
@angelnu I force pushed https://github.com/MetacoSA/terraform/pull/2/commits/81e1db7f9552aaad20b685fd2cb681d1173720c4 , could you please review ?.. Thanks

**Pull request merged**